### PR TITLE
Add docker image with root user.

### PR DESCRIPTION
**Pull request by luca-heltai**
This allows one to use our docker image directly in github actions:

```
name: Run on dealii/dealii:master-focal-root
on:
  push:
    branches: [ main ]
  pull_request:
    branches: [ main ]

jobs:
  continuous-integration:
    runs-on: ubuntu-latest
    container: dealii/dealii:master-focal-root
    steps:          
      - name: Checkout code
        uses: actions/checkout@v2
      - name: Build and run tests
         run: |
            mkdir build
            cd build
            cmake -GNinja .. 
            ninja 
            ninja test
```

### CODE REVIEW

**Comment by luca-heltai** [comment]
No changes in code here. This will only be triggered once merged to master.

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -27,4 +27,14 @@ jobs:
...
+
+      - name: Build and push Docker image of master with root user
+        uses: docker/build-push-action@v2
+        with:
+          context: ./contrib/docker/
+          file: {context}/Dockerfile.root
+          cache-from: type=registry,ref=dealii/dealii:master-focal
+          cache-to: type=inline
+          push: true
+          tags: dealii/dealii:master-focal-root
```
Comment (marcfehling) [inline .github/workflows/docker.yml:40]: If you think it's necessary, you can also add a newline here.

**Review by marcfehling — COMMENTED** [review]
I'm not familiar with docker, but this looks good to me.

Does your script automatically create the `master-focal`/`master-focal-root` tags?

**Comment by luca-heltai** [comment]
@marcfehling , yes. The yaml file also creates the new tag `dealii/dealii:master-focal-root`.

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
Just for completion: Instead of using the `master-focal-root` user image, you can also override the default user in the docker image like this.
```
image: dealii/dealii:master-focal
options: --user root
```

See also https://github.com/dealii/code-gallery/pull/93#issuecomment-1500775707

**Comment by luca-heltai** [comment]
Nice @marcfehling. We should revert this then. I didn't know about the `options:` option.

**Pull request merged**